hw5-10_to_do_list/oop/js/repositories.js

```diff
@@ -0,0 +1,47 @@
+class TaskRepository extends Repository {
+    constructor({ storageHost, storagePort }) {
+        super()
+        this.storageURL = `http://${storageHost}:${storagePort}`
+        this.taskEndpoint = `${this.storageURL}/tasks`
+    }
+
+    async getAll() {
+        const response = await fetch(this.taskEndpoint)
+        return response.json()
+    }
+
+    async add(entity) {
+        const response = await fetch(this.taskEndpoint, {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(entity),
+        })
+        return response.json()
+    }
+
+    async update(entity) {
+        const url = `${this.taskEndpoint}/${entity.id}`
+        await fetch(url, {
```

Comment:
```suggestion
        return fetch(url, {
```
It is not implemented by you and out of HW scope but in real life you need to check response status in the caller of this method to be sure that the task was really updated on the server.